operate/client/src/App/Decisions/InstancesTable/index.tsx

```diff
@@ -6,98 +6,96 @@
-import {groupedDecisionsStore} from 'modules/stores/groupedDecisions';
 import {formatDate} from 'modules/utils/date';
-import {useEffect} from 'react';
-import {useLocation} from 'react-router-dom';
 import {Container, DecisionName} from './styled';
 import {observer} from 'mobx-react';
 import {Paths} from 'modules/Routes';
 import {tracking} from 'modules/tracking';
 import {Link} from 'modules/components/Link';
-import {useFilters} from 'modules/hooks/useFilters';
-import {getDecisionInstanceFilters} from 'modules/utils/filter';
+import {useDecisionInstancesSearchPaginated} from 'modules/queries/decisionInstances/useDecisionInstancesSearchPaginated';
+import {
+  useDecisionInstancesSearchFilter,
+  useDecisionInstancesSearchSort,
+} from 'modules/hooks/decisionInstancesSearch';
 
 const ROW_HEIGHT = 34;
+const SMOOTH_SCROLL_STEP_SIZE = 5 * ROW_HEIGHT;
 
 const InstancesTable: React.FC = observer(() => {
+  const filter = useDecisionInstancesSearchFilter();
+  const sort = useDecisionInstancesSearchSort();
+  const location = useLocation() as Location<{refreshContent?: boolean}>;
+
   const {
-    state: {
-      status,
-      filteredDecisionInstancesCount,
-      latestFetch,
-      decisionInstances,
+    data,
+    status,
+    isFetching,
+    refetch,
+    isFetchingPreviousPage,
+    hasPreviousPage,
+    fetchPreviousPage,
+    isFetchingNextPage,
+    hasNextPage,
+    fetchNextPage,
+  } = useDecisionInstancesSearchPaginated({
+    payload: {filter, sort},
+    enabled: filter !== undefined,
+    select: (data) => {
+      tracking.track({
+        eventName: 'decisions-loaded',
+        filters: Object.keys(filter ?? {}),
+        ...sort[0],
+      });
+      return {
+        decisionInstances: data.pages.flatMap((page) => page.items),
+        totalCount: data.pages.at(0)?.page.totalItems ?? 0,
+      };
     },
-    areDecisionInstancesEmpty,
-    hasLatestDecisionInstances,
-  } = decisionInstancesStore;
-
-  const location = useLocation();
-  const filters = useFilters();
-
-  const {isInitialLoadComplete} = groupedDecisionsStore;
-
-  useEffect(() => {
-    const params = new URLSearchParams(location.search);
-    const decisionId = params.get('name');
-    const tenantId = params.get('tenant');
-
-    if (isInitialLoadComplete && !location.state?.refreshContent) {
-      if (
-        decisionId !== null &&
-        !groupedDecisionsStore.isSelectedDecisionValid({
-          decisionId,
-          tenantId,
-        })
-      ) {
-        return;
-      }
-
-      decisionInstancesStore.fetchDecisionInstancesFromFilters();
-    }
-  }, [location.search, isInitialLoadComplete, location.state]);
+  });
+  const decisionInstances = data?.decisionInstances ?? [];
+  const filteredDecisionInstancesCount = data?.totalCount ?? 0;
 
   useEffect(() => {
-    if (isInitialLoadComplete && location.state?.refreshContent) {
-      decisionInstancesStore.fetchDecisionInstancesFromFilters();
+    if (location.state?.refreshContent) {
```

Comment:
@christoph-fricke Do you have any idea why this is necessary?

Reply:
@vsgoulart No idea. Without it, data is not refetch when an active navigation link is clicked again. In other words, it acts as an implicit "refetch view data" feature... 🤔